Seminars/Sem.02/Pract.02/Task7/Task7.cpp

```diff
@@ -0,0 +1,238 @@
+        index++;
+    }
+
+    user.password[passwordIndex] = '\0';
+}
+
+void writeSystemOfUsersToFile(const System& users, const int usersCount)
+{
+    std::ofstream usersFile("users_system.txt", std::ios::app | std::ios::_Nocreate);
+
+    if (!usersFile.is_open())
+    {
+        std::cout << "Error ocurred when trying to open the file." << std::endl;
+
+        return;
+    }
+
+    for (size_t i = 0; i < usersCount; i++)
+    {
+        usersFile << users.users[i].name << ',' << users.users[i].email << ',' << users.users[i].password << '\n';
+    }
+
+    usersFile.close();
+}
+
+void readSystemOfUsersFromFile(System& users, int usersCount)
+{
+    std::ifstream usersFile("users_system.txt", std::ios::app | std::ios::_Nocreate);
+
+    if (!usersFile.is_open())
+    {
+        std::cout << "Error ocurred when trying to open the file." << std::endl;
+
+        return;
+    }
+
+    int line = 1;
+
+    while (!usersFile.eof())
+    {
+        char buffer[MAX_BUFF_SIZE];
+        usersFile.getline(buffer, MAX_BUFF_SIZE);
+
+        if (line == 1 || line - 2 == usersCount)
+        {
+            line++;
+            continue;
+        }
+
+        readUserFromFile(buffer, users.users[line - 2]);
+        line++;
+    }
+
+    toBeginningOfFile();
+}
+
+void StrCopy(char* dest, const char* source)
+{
+    size_t index = 0;
+    while (source[index] != '\0') {
+        dest[index] = source[index];
+        index++;
+    }
+
+    dest[index] = '\0';
+}
+
+int countLines()
+{
+    std::ifstream usersFile("users_system.txt", std::ios::app | std::ios::_Nocreate);
+
+    if (!usersFile.is_open())
+    {
+        std::cout << "Error ocurred when trying to open the file." << std::endl;
+
+        return -1;
+    }
+
+    unsigned lines = 0;
+
+    while (!usersFile.eof())
+    {
+        char buffer[MAX_BUFF_SIZE];
+        usersFile.getline(buffer, MAX_BUFF_SIZE);
+        lines++;
+    }
+
+    toBeginningOfFile();
+
+    return lines - 1;
+}
+
+
+int main()
+{
+    char command[MAX_BUFF_SIZE];
+
+    System users;
+    int numOfUsers = countLines();
+    int currentUserIndex = countLines();
+
+    do
+    {
+        std::cout << "Enter command:" << '\n' << "\\>";
+        std::cin >> command;
+
+        if (strcmp(command, "register") == 0)
+        {
+            User user;
+            std::cout << "Enter name:" << '\n' << "\\>";
+            char name[MAX_LENGTH];
+            std::cin >> name;
+            StrCopy(user.name, name);
+
+            std::cout << "Enter email:" << '\n' << "\\>";
+            char email[MAX_LENGTH];
+            std::cin >> email;
+            StrCopy(user.email, email);
+
+            std::cout << "Enter password:" << '\n' << "\\>";
+            char password[MAX_LENGTH];
+            std::cin >> password;
+            StrCopy(user.password, password);
+
+            writeUserToFile(user);
+
+            std::cout << "Registration successful" << '\n';
+        }
+        else if (strcmp(command, "login") == 0)
```

Comment:
You rely on the fact that the user will not use commands register and then login. Because of the fact that you do not change the number of current users , when you save new registered user into the file, then you cannot login with his credentials